osquery/logger/plugins/aws_util.h

```diff
@@ -77,20 +79,42 @@ class OsqueryFlagsAWSCredentialsProvider
 };
 
 /**
+ * @brief AWS credentials provider that uses STS assume role auth
+ *
+ * This provider deligates temp AWS STS credentials via assume role
+ * for and AWS arn.
+ */
+class OsquerySTSAWSCredentialsProvider
+    : public Aws::Auth::AWSCredentialsProvider {
+ public:
+  OsquerySTSAWSCredentialsProvider() : AWSCredentialsProvider() {}
+
+  Aws::Auth::AWSCredentials GetAWSCredentials() override;
+  Aws::String sts_access_key_id = "";
+  Aws::String sts_secret_access_key = "";
+  Aws::String sts_session_token = "";
+  long long token_expire_time = 0;
```

Comment:
Actually it looks like we use `size_t` throughout the code base when we want unsigned values to hold times.